### ISSUE TITLE
Add `*` parsing to TeX root search

### ISSUE THREAD

**Issue by 927589452** [lervag/vimtex#1482]
**Is your feature request related to a problem? Please describe it.**
Sometimes one knows where a `main.tex` file will be relative to the included file

**Describe the solution you'd like**
Add glob parsing to the root search

**Describe alternatives you've considered**
An alternative would be to add the NONE directive and specify each root document with a `.latexmain`

**Additional context**
Example files will be included in PR

**Comment by lervag** [comment]
Sorry, but I don't think this feature request is quite complete. In the example files, none of the include files are included in the `main.tex` file. Could you please:
1. Explain or give an example of a proper `mian.tex` file.
2. Update the original post with steps to test for correct implementation. Preferable with "expected" vs "observed" behaviours.

**Comment by 927589452** [comment]
> Sorry, but I don't think this feature request is quite complete. In the example files, none of the include files are included in the `main.tex` file. Could you please:

That's the idea; these are files, that are connected to the project, but not directly input into it;
this might be a file generating a file that is included (for example by `bib2gls`) or a `latexmk` rule

>     2. Update the original post with steps to test for correct implementation. Preferable with "expected" vs "observed" behaviours.

If I define a `% !TeX root = ../*.tex`, `vimtex` should ask for the correct file like it does with the NONE option, if there is more than one file and select the matched if there is only one match

**Comment by lervag** [comment]
Ok, I think I begin to understand. You propose two things:

1. Support for glob expressions in the `TeX root` directive, which will ask if more than one candidate is found.

2. Support for a `NONE` value for the `TeX root` directive. However, I don't understand what the point is with this one. You write in the first post that it means that one should look for the `.latexmain` file specifier, but then you don't need the `TeX root` directive?

**Comment by 927589452** [comment]
>     1. Support for glob expressions in the `TeX root` directive, which will ask if more than one candidate is found.

Exactly
>     2. Support for a `NONE` value for the `TeX root` directive. However, I don't understand what the point is with this one. You write in the first post that it means that one should look for the `.latexmain` file specifier, but then you don't need the `TeX root` directive?

you already implemented the `NONE` directive on my request ^^ #1440

**Comment by lervag** [comment]
Ah, I see. So only 1. is necessary, since 2. is already implemented and working?

Do I understand correctly that the `NONE` value is vimtex specific, and so it should be documented as well? It seems it is not documented at the moment.

**Comment by 927589452** [comment]
yes, 2  works very well if the `.latexmain` is used

**Comment by lervag** [comment]
Ok, I think this is fixed now.

Note: The update to the docs in 18663dd was actually wrong. The feature you describe as `TeX Root = NONE` is not related to the TeX Root directive. Or, instead, the feature implemented in #1440 was not related to the TeX Root directive, but instead an improvement to the recursive strategy in how it handled multiple candidates. For this to work, there is no reason to have the `TeX Root = NONE` directive.

Except, perhaps your point is that you want this directive added? It could be useful e.g. in the following case:

**main.tex**
```tex
\documentclass[draft,debug]{article}

\begin{document}
some more stuff
\end{document}
```

**included.tex**
```tex
%! TeX Root = NONE

Stuff to be included.
```

Note: `included.tex` is _not_ included in main.tex.

If we do:

1. Open `main.tex`
2. `:edit included.tex`

Then the _current_ behaviour is for `included.tex` to become its own project. Perhaps you are suggesting that the `TeX Root = NONE` should indicate to always inherit so that after step 2, the main file is still `main.tex`?
